### ISSUE TITLE
Explicit procedure for bad behaviour

### ISSUE THREAD

**Issue by h-lame** [lrug/lrug.github.io#27]
Right now we say that we'll "step in" when things go awry on the mailing list, but that clearly isn't enough as people just keep emailing their snark.  We should have an explicit rule about putting offenders on moderation, and then ultimately removing them from the list if they keep offending.

We already moderate new members and we should have an explicit rule about when to remove moderation from new members or people who have ben re-moderated.

**Comment by h-lame** [comment]
Of course, a mild problem with this process is that people on moderation who keep breaking the code of conduct will likely only be seen by the moderation team.  It feels somewhat problematic to put the burden of moderation => ejection vs. moderation => acceptance on the moderation team and for that to happen in an effectively private space.

**Comment by tef** [comment]
A mild problem is better than the gigantic one you currently face: You have no means of moderating the community you are responsible for. It feels somewhat problematic to dump assholes on the community, and ask them to convince the asshole to leave.

If you offer an open invitation to a public event, there is an onus on you to moderate the community. Although you offer a code of conduct, it is theatre without enforcement. 

fwiw: usual scale is this:

- public warning from a moderator (for minor and accidental infractions)
- full moderation of posts for a month (for major infractions)
- full ban (abuse, harrassment, spam)

**Comment by tomstuart** [comment]
Notes:

* “LRUG” is (roughly) two separate things: the mailing list, and the monthly meetings.

* A meeting is an event, and has clearly visible authority figures (host, Skills Matter staff, pub landlord etc) along with an obvious enforcement procedure — if you do something bad, you will ultimately be reported to one of those authorities, and they will throw you out. The current code of conduct makes this reasonably clear; it could be more explicit about the exact sequence of events, but it already has a reporting procedure, and it’s not obvious that spelling out the enforcement part would necessarily add useful detail. Although the past is not a predictor of the future, previous incidents at meetings have been handled with apparent success, so the current regime is at least not a total wash-out.

* The mailing list is more difficult because it’s not an event, and nobody’s in charge; “the community” is all there is, so “the community” is all that’s available to police it; the question is how. The list is already moderated, but I don’t know by exactly whom (@h-lame, @lazyatom, …?) or according to exactly what criteria, although I believe historically it’s been more about broad filtering of recruiter and/or automated spam than policing exclusionary behaviour. It seems like it would be beneficial to make this process more formal and transparent, perhaps to the extent of documenting who moderates the emails and on what basis they will be moderated, while accepting that these people can only ever make a best-effort attempt to police the actual content — with the best will in the world, it’s essentially impossible for anyone to spot 100% of potentially offensive behaviour, so it should be clear that they are providing a first line of defence without necessarily guaranteeing the safety (or approving the content) of everything that makes it through.

* We should collectively accept that this is a tricky problem with a non-obvious solution, and not try to either ignore it or sweep it under the convenient dismissal of “well-obviously”; none of this is straightforward, and will require further effort to get right. We are all trying our best, and there have been ample opportunities for experienced folk to contribute in the past (e.g. #13), yet here we are with plenty of work still to do.

**Comment by rdrake98** [comment]
From a randomly sporadic lurker till he tried to 'help out' two days ago.

Most important for me is Tom's last point. There needs to be a health warning up front on any new code of conduct/procedural guidelines: "Taking part in, or just reading, the LRUG mailing list will not resolve all problems you have with {race relations} in London or anywhere else in the world." Insert pretty much anything you like in the curly brackets. I had that preface in mind before reading Tom so he can't be blamed.

Vish Vishvanath said yesterday "There's plenty of support against sexism in the community, and the stories are pretty well-known. Perhaps the stories around racism aren't as well-documented, or happened a long time ago, or just aren't as outrageous any more." Is it possible to point to the stories he means so they become even better-known? (I wouldn't have a clue. All the events I've attended I think have been first class. But nothing untoward happened, that I noticed anyway. And that's your genius guys.)

I think distinction has to be maintained between "I found this annoying", "I found this offensive" and "this was racist". Of course sometimes all three are indisputably true, in which case the offending piece is disappeared. On Tuesday I didn't think the last statement applied. I bowed out of the public debate and vowed to reflect on it. By the next day I was more convinced it wasn't. This kind of thing will happen. People will disagree. One big rule I would assert is that the most offended don't automatically win. But you have to listen if people are offended. No easy answers.

Would it make sense to make moderation more of a goldfish bowl where a larger group is able to see the bad stuff that hits the cutting room floor? So they can question, advise and reflect?

Am I being moderated now? One never knows.

**Comment by rdrake98** [comment]
Or a more literary preface:

How small, of all that human hearts endure
That part that laws or kings can cause or cure

--  Samuel Johnson

**Comment by aanand** [comment]
Agree with @tomstuart and @tef:

1. A graded scale of responses from warning -> temporary mute -> full ban.
2. Document the moderation policy and refer to it whenever taking action.

**Comment by rdrake98** [comment]
Can I repeat my suggestion (made to @h-lame and @lazyatom by email) that this specific incident, and everything people feel should have happened or should not have happened, within it, be openly debated by interested LRUG parties but not on the LRUG mailing list. To be a bit more specific about that:

* 'interested LRUG parties' would not include new members or those still in moderation
* the discussion would be password-protected, invisible to anyone not signed up for it.

I also suggested to James and Murray that it might be good to leave it for up to month before starting such a debate. I think it's particularly important to get the candid input of women like Despo, without them feeling the debating style is too macho to want anything to do with it. But for woman perhaps read sensitive, thoughtful person. Not an easy one to kick off and get right. But if it did go off the rails at least it wouldn't have directly polluted the main LRUG mindspace.

Such a discussion could I think shed a lot of light on what a next version of the code of conduct, that specifically and clearly applies to the mailing list rather than events, should look like. I think Tom's given good reasons why it has to be different and I think there may be others. But I'll keep any or all of those thoughts to that forum, if it comes to exist.

**Comment by tomstuart** [comment]
I can’t see any reason to make such a discussion private. The only consideration is that it’s not forced upon anyone who doesn’t to participate (so, yes, the LRUG chat list is not a good venue), but privacy isn’t a prerequisite for that.

I personally am less interested in discussing this specific most recent event, and more interested in working out what steps are necessary to prevent other upsetting things from happening in future.

I don’t have a great deal to say; the content of @aanand’s comment is a good summary of what I think should happen.

**Comment by aanand** [comment]
@rdrake98: I agree with @tomstuart that there really isn’t much to discuss about how that incident went down – for what it’s worth, though, here’s my take on what shouldn’t have happened:

- A recruiter shouldn’t have made a racist joke at the expense of Asian developers.
- You shouldn’t have attempted to educate someone *at whose expense that joke was made* on how to respond to racism.
- You shouldn’t have attempted to downplay the racism by comparing it to other, unrelated acts of racism.
- You shouldn’t have signed off with a passive-aggressive “there’s more to learn elsewhere” and a smiley face.

If you want to start a forum to discuss how unfairly you were treated, that’s your business. I don’t think it’s LRUG’s business in the slightest.

**Comment by rdrake98** [comment]
@aanand: You're not the first person to have wrongly characterised what I am doing and I mean wrongly characterise for certain. (Some things are definitely a matter of interpretation. I won't have time for everything now.)

I didn't want to start a forum to discuss how unfairly I was treated. I would of course use it to put my point of view but that's a very different thing. When I emailed on Thursday, instead of making any further defence on the list, Murray pointed me here but I thought "this quiet place can't be it". And now I think perhaps it is! I am happy to go with whatever flow there is. The privacy point was not crucial, I bow to Tom and you on that.

Obviously we disagree about whether the 'joke' (it hardly was that, more an attempted semi-jocular turn of phrase) was racist. But does that matter? The code of conduct, which I take to be total in its authority for LRUG *meetings*, says "if someone feels harassed or excluded by your words or actions, then those words or actions constitute harassment or exclusion. Your intent is not a factor."

Anyway. What would have been evil would have been to name the country concerned and say it was a shithole. (One of a million possible evil things we would never have seen thanks to the moderators.) I say evil because racism is just one part of hating people different to you. All of that is evil. It doesn't need to be race that triggers that. I'm sure we'd agree with that.

The author didn't want to name the country concerned, quite rightly, but he did want to communicate that it was dirt poor. I thought at once of Tajikistan, because I have friends who worked there for many years in a charity trying to help out practically. In the 1990s it was regarded as the poorest of the -stans and this was confirmed at a meeting I went to at the European Bank for Reconstruction and Development with my friends and some Tajik experts. The people we heard about from them seemed like heroes from another universe. I hope today, with all the developments in the region, that there are many Tajik men and women learning to program and on the LRUG list.

I don't buy then that the awkward phrase the author ended up using was a "racist joke at the expense of Asian developers". If it was an obvious one it would have been picked up by the moderators and we would have never have seen it, right? 

But let me cut to the end game. I wouldn't object to the whole thread in question being deleted, even given my present position. If it was a racist joke then I think the whole thread definitely should be deleted. If the code of conduct says, in effect, that my opinion means nothing because someone felt harassed and excluded by it then should it be deleted? Not sure. You tell me.

**Comment by rdrake98** [comment]
I take it that all readers think there's no point in a debate here. I assume they feel there's been enough debate already, on the list, here and by email. Here's one thing that puzzles me about this. In my first contribution here six days ago I wrote:

> Vish Vishvanath said yesterday "There's plenty of support against sexism in the community, and the stories are pretty well-known. Perhaps the stories around racism aren't as well-documented, or happened a long time ago, or just aren't as outrageous any more." Is it possible to point to the stories he means so they become even better-known?

Nobody has attempted to do so. Or even said why it's not appropriate to do so. Why is that?

I'm sure I've not been party to all the debate elsewhere and I can live with that. But the public record at it stands strikes me as highly deficient. Let me lend a hand in this post to those that disagree with me and think the original post on 17th was racist. This is how I assume you reason about it. (I am having to guess. There's been no reasoning at all that I've seen. That's not good enough. I'll come back to that point at the end.)

In saying

> We started out thinking (against what turned out to be great advice) that we could hire a cheap consultant from (insert -STAN country) to build our product for £50

Stefan Surzycki was being racist because he was implying (multiple choice)

1. All -stan countries are exactly the same
2. All developers from -stan countries are shit
3. All -stan countries are shitholes which couldn't produce a decent developer if they tried
4. One particular developer from a -stan country wasn't as good as they'd hoped and this reflects badly on all -stan countries and the people that come from them.

The fourth seems to me the strongest or nearest to being plausible. But it still very, very implausible that this is what the author thought in his heart and thus expressed, inadvertently, in his words. The other three are way beyond very, very implausible.

The first obvious thing to say in the author's defence - something I said in my first contribution at the time - is that he and his partner were taking primary responsibility for the mistake they'd made with their startup so far. The person from an unknown -stan country involved for a while was not, in my reading, being blamed at all. There were communication problems and they, the management consultants, were to blame for making unrealistic assumptions. This, and their frank admission of it, is a credit to them.

But they are racists, we know that because Najaf Ali told us so. I'm sorry I spoke.

Actually I'm not. I just wanted to show I could do irony too.

What I am attacking here is not any individual but the idea that a mere accusation of racism from any POC is enough to settle the question forever. Either this question was settled at 16:42 on 17th February or it wasn't. In my view it wasn't. The fact that someone like myself could question it and never have good reasons presented to them why the post was racist is a travesty of natural justice. Not to me. This has never been about me, it's about whether LRUG treated Stefan Surzycki fairly and the implications of that for the future.

It's also about how to de-escalate tensions that are bound to arise from time to time. According to Murray Stefan opted not to apologise for any offence he caused - or at least Murray doesn't know of any such apology (and I find it highly unlikely that any such apology would have taken place without Murray knowing, bless him and his fellow moderator). Put yourself in Stefan's shoes. You're thinking of making an apology for something you've said that has inadvertently caused offence. Is escalation of your offence to 'horrific racism' - I'm putting together two posts there - going to help you apologise? No. For very good reasons such escalation would make anyone hesitate, lest "Sorry I caused offence" is taken to mean "Sorry I am a horrific racist."

I'm sure all of this has been discussed in emails I've not been party to but I still find it strange there's been no public expression of these very basic points.

**Comment by rdrake98** [comment]
Here's another argument that might I guess be advanced by those convinced Stefan's post was racist: that the term '-stan' is in itself already well-established as a racist slur.

If this is true I am wholly ignorant of it as a lifelong English speaker. But there are many things of which I am wholly ignorant. What's strange, though, is that nobody thought to make this point to me and link to relevant discussions of where this abhorrent term has been used in this way.

To illustrate what I'm talking let's say Stefan had said

> We used a young Afghani who was willing to accept a low rate for a while but unfortunately it didn't work out.

Because 'Afghani' I don't believe is ever taken as pejorative in itself this wouldn't be racist. If he'd said:

> We used a young Kurdi who was willing to accept a low rate for a while but unfortunately it didn't work out.

this could be taken as something between cute and careless, because Kurd is the normal term and this is not common usage. And if the programmer had been from Pakistan and he'd tried exactly the same usage his post would have been racist. I and no doubt everyone on LRUG would agree.

So where is the evidence that the term -STAN in itself is remotely like this latter case?

**Comment by tomstuart** [comment]
Richard, your comments are derailing the issue, which is “how should LRUG handle bad behaviour?”, not “what is bad behaviour?”.

**Comment by rdrake98** [comment]
Hi Tom, bad behaviour could mean anything from accidental insensitivity to 'horrific racism'. Which one it is has quite a bearing on how it should be handled. But here's my current position. If the moderators let something through that someone else thinks is racist the whole thread is suspended (ie made invisible) until such time as the accusation of racism is either agreed* or not. If it is agreed the post never reappears. If it isn't attempts are made to effect reconciliation between the original poster and the person(s) who have objected. Depending on how that goes the post may reappear. What isn't on is for dubious accusations of racism against someone using their real name, who is basically unknown to LRUG, to persist via the archive for the rest of time.

**Comment by rdrake98** [comment]
* The agreement should be unanimous, of the wider group of watchers and moderators I mooted earlier. Elections for these people might not be a bad idea. I depends how far one wants to take this.

**Comment by Najaf** [comment]
Slightly off-topic but are the contents of this issue discussion and the relevant LRUG thread in the public domain? The reason I ask is that I've found the whole experience very educational, so I'm thinking of putting together a workshop, something like:  *Playing all the cards - how to talk to the natives so they know their place*.

**Comment by tomstuart** [comment]
Disappointingly, this issue has already degenerated into an example of the very problem it’s trying to address: noise, name-calling, and general denial-of-service of the time and attention of people who wish to contribute positively.

A community is more than just an arbitrary collection of people; it’s a collection of people who come together with some unifying interest or purpose, whether that’s geographical proximity, a particular hobby, or whatever.

LRUG is a London-based community for people who are interested in the Ruby programming language. In order to broadly appeal to everyone who shares that interest, LRUG’s remit must be correspondingly narrow: the meetings and the mailing list are for talking about the Ruby programming language and closely-related topics. That way, people who decide to come to an event or sign up for the list can be confident that they know what they’re going to get.

By choosing to participate in LRUG, each member has a responsibility to not expose other people in the community to anything they weren’t expecting. The meetings and the mailing list are not a space for general socialising, “edgy” humour, political discourse, personal attacks, harassment, recruiter baiting, discriminatory behaviour, or anything else essentially unrelated to Ruby.

That doesn’t make it a police state, merely a focused and purposeful community. The ongoing viability of LRUG is absolutely contingent upon this focus and purpose. Any member who feels unfairly constrained by LRUG’s purpose is completely free to create another space in which the aforementioned behaviours are part of the status quo; people who are comfortable with that environment will come along and participate, and those who aren’t will stay away.

So the bar for what constitutes “bad behaviour” must necessarily be low, but that’s fine, because _it’s easy to talk about Ruby without upsetting anyone_. If you’re unable to contribute to a discussion about Ruby without denigrating someone else or making a joke that could be interpreted as racist, then your contributions belong on your own blog or event or mailing list, not at LRUG.

It is therefore literally irrelevant whether an apparently racist joke is “really” racist. If it even raises the question, it is unacceptable; there is no need for a jury to decide the validity of an accusation. As & when this process errs, it will err on the side of safety, inclusivity and focus, which is absolutely desirable.

It is exhausting to constantly have to re-explain this fairly basic idea; even though [the mailing list signup page](http://lists.lrug.org/listinfo.cgi/chat-lrug.org) explicitly says “remember that "humour" can be easily misread in a text only medium so err on the side of being clear”, it doesn’t seem as though anyone reads it. That’s why we need an explicit enforcement procedure: to proactively communicate that information to people who are otherwise unable or unwilling to understand what LRUG is for.

**Comment by samsworldofno** [comment]
One thought I had on this topic was retiring the mailing list and replacing it with a forum. This could potentially better segregation of posts (rather than remembering [JOBS], you'd just post in the right room), easier archive access and lots of moderation facilities. I didn't know that the mailing list was moderated, but I know it's not possible to close a thread (and even then, the dreaded replies to the digest still flow in) and certainly the moderation is not publicly visible/auditable.

Obviously just a switch will not solve anything/everything, but perhaps it will enable more active moderation. There are great articles online from the Stack Overflow people on the moderation approaches - perhaps not all agreeable to everyone, but they have managed to create an effective community. 

I also think a forum would be more accessible. The rules of email (particularly top/bottom posting) are not clear to beginners and the lack of being able to edit/correct your posts can be daunting. The web is a way more accessible format just by virtue of its operation, I'd say.

The other bonus of forum over email is the software is more up-to-date in the form of Discourse. We won't be back in the phpBB stone ages like we would have been a few years ago if we had considered this change.

Perhaps this has been mooted or tried before?

**Comment by rdrake98** [comment]
I agree with the three latest contributors. As far as I am concerned there's no IP to protect in what I've written but I do have the concern already expressed that a named third-party has been accused of horrific racism. I think that stuff should be deleted from LRUG the archive - and at the least anonymised if quoted in any other blog post or discussion. I don't mind any criticism of myself remaining. I'd prefer to be quoted verbatim with proper context provided of course!

Tom's right in everything he says about the future. The moderators must wish they'd spotted this problem in advance. Once over five people were launching attacks on a single named individual who was new to LRUG I thought that was very ugly indeed. I'd like to put on the record my appreciation for Matthew Rudy Jacobs who alone of any LRUG person tried to welcome the newcomers. Ignore the trolls was the wrong way to start but the rest was terrific. This hasn't been said yet. Perhaps I alone of subscribers want to say it but I do and now I have!

Sam's right to ask the question about newer technology but I think the more relevant points are in what Tom said. How does one get new posters to read the the mailing list signup page?

Lastly I'd like to go right off topic (for the list, that is, but not I think for this thread) and tell the story of my week. Since the 70th anniversary of the liberation of Auschwitz on 27th January I'd resolved to watch all the BBC programmes on the subject before the iPlayer month's deadline expired. On Tuesday I was stymied on this because I was travelling and my iPhone went on the blink so I wasn't able to finish the four and half hours of Shoah part 1 (not an easy watch at the best of times). The next day I drove up to the Apple Store at Cribbs Causeway outside Bristol for a session at the Genius Bar. I was a bit early so I added the post about whether -stan has automatic stigma here. I took my place in the queue at the appointed time and it turned they were running about half an hour late, so I got talking to the lady besides me. She was a good talker, about 60 I'd say, very keen on her Apple iPad but with a problem with her email. Almost the first thing she wanted to tell me about was her friend from Nigeria who was much more technical than she was but hadn't been able to solve the problem on this occasion. This Nigerian lady had become a firm friend - "we just get on so well together". It happened to be her 40th birthday that very day and she had broken down and cried when so many people she'd met had remembered this big day for her in such a strange country.

This POC who I assume I will never meet has nevertheless become a major inspiration for me. We can do better. I thanked God for this very direct encouragement to that end.

**Comment by aanand** [comment]
Richard, please stop posting. Your contributions to this discussion have been worse than useless.

It does not matter that a person got accused of racism. It does not matter that that discussion remains on public mailing list archives. Future generations will draw their own conclusions.

It matters that a person made a joke that was interpreted by some people as racism targeted at them. That's the "exclusionary behaviour" Tom is talking about, and that's what LRUG moderators need to take quicker action on in future.

**Comment by rdrake98** [comment]
Aanand, our shared behaviour deeply influences our future culture. We'd all agree I'm sure that quicker action should be taken in future but in this case the toothpaste was out of the tube very quickly and for me it counts what statements are made through praxis as well as the stories we tell about this incident from here on in. These are the typical ingredients of any culture. The issues cannot be ducked.

**Comment by aanand** [comment]
![plz-stop-post](https://cloud.githubusercontent.com/assets/1062/6419459/3ba51b90-beb5-11e4-80fa-73f2d2b7d60c.jpg)

**Comment by Najaf** [comment]
Oh no, keep going. I for one am a great fan of Richard's posts and am going wild in anticipation for what he produces next.

**Comment by rdrake98** [comment]
![Shut Up](http://ecx.images-amazon.com/images/I/51S9PXHC6NL.jpg)

I have just one more thing to explain, as it turns out, and I'll be done. By tomorrow.

**Comment by h-lame** [comment]
We’re locking this issue for now as the discussion has become derailed from the original purpose.  

This is not to say we're ignoring the problem; we are working on changes to the code of conduct to attempt to address the original questions raised in this issue.  We'll report back shortly in a new PR where we encourage interested parties to have a discussion based on those changes and how they could be improved.
